src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceGenerator.cs

```diff
@@ -45,6 +47,8 @@ public sealed class SourceGenerator : ISourceGenerator
             defaultSeverity: DiagnosticSeverity.Error,
             isEnabledByDefault: true);
 
+        private static Tuple<ImmutableArray<byte>, ImmutableArray<(string hintName, SourceText sourceText)>> s_cachedResult;
```

Comment:
consider doc.

Reply:
consider a nullable c#-tuple  could name these members then instead of Item1/Item2.

Reply:
Reference types ensure atomic reads/writes. I have a note regarding atomicity on the assignment, but I could switch to a helper method that contractually ensures atomicity (e.g. `Volatile.Write`).

Reply:
Switched to `Volatile` for the operations, and extracted the caching logic to a helper base class.